buildSrc/subprojects/binary-compatibility/src/main/groovy/gradlebuild.binary-compatibility.gradle

```diff
@@ -141,12 +147,30 @@ def checkBinaryCompatibility = tasks.register("checkBinaryCompatibility", Japicm
         excludedClasses = toPatterns(PublicApi.excludes + PublicKotlinDslApi.excludes)
 
         title = "Binary compatibility report for Gradle ${isSnapshot.get() ? "${baseVersion.get()}-SNAPSHOT" : version} since ${compatibilityBaselineVersion}"
-        destinationDir = file("$buildDir/reports/binary-compatibility")
-        reportName = "report.html"
+        destinationDir = binaryCompatibilityReportDir
+        reportName = binaryCompatibilityReport
     }
 
     BinaryCompatibilityHelper.setupJApiCmpRichReportRules(delegate, acceptedViolations, apiSourceFolders, baseVersion.get())
 }
+
+gradle.taskGraph.afterTask {
+    if (it.state.failure != null && it instanceof JapicmpTask) {
```

Comment:
You can set the description for the rich report on the `JapiCmpTask`. Is that enough? https://github.com/gradle/gradle/blob/00bed93d16a05c6f0574f2b96694ffa79e136573/buildSrc/subprojects/binary-compatibility/src/main/groovy/gradlebuild.binary-compatibility.gradle#L139-L146

Reply:
Yes.